src/pages/mypage/profile.tsx

```diff
@@ -1,34 +1,101 @@
-import { useState } from 'react';
+import { useMemo, useState } from 'react';
+import { useUserQuery } from '../../api/hook/users';
 import Button from '../../components/common/button';
 import TextInput from '../../components/common/form/text-input';
 import HomeLayout from '../../components/common/layout/layout';
 import { ResignButtonContainer, Section, StyledForm, Title } from '../../styles/mypage/profile';
+import { userIdState } from '../../recoils/atoms/users';
+import { useRecoilValue } from 'recoil';
+import { patchUserInfo, nicknameCheck } from '../../api/fetcher/users';
+import NicknamePopup from '../../components/mypage/mypage-pop-up';
+import { useMutation, useQueryClient } from 'react-query';
+import _, { debounce } from 'lodash';
+
+interface errorMesageType {
+  unknown: string;
+  isError: string;
+  notError: string;
+  [prop: string]: string;
+}
 
 const Profile = () => {
-  const [user, setUser] = useState({
-    id: '1',
-    email: 'user@email.com',
-    nickname: '기존닉네임',
+  const [nickname, setNickname] = useState<string>('');
+  const [error, setError] = useState<string>('');
+  const [nicknamePopupOpen, setNicknamePopupOpen] = useState<boolean>(false);
+
+  const userId = useRecoilValue(userIdState);
+  const queryClient = useQueryClient();
+
+  const { data } = useUserQuery(userId as number, {
+    onSuccess: data => {
+      setNickname(data.nickname);
+    },
+  });
+
+  const errorMessage: errorMesageType = {
+    unknown: '닉네임을 입력해주세요',
+    isError: '이미 등록된 별명입니다',
+    notError: '',
+  };
+
+  Object.freeze(errorMessage);
+
+  const { mutate: patchUserInfoMutate } = useMutation(patchUserInfo, {
+    onSuccess: () => {
+      setError('notError');
+      setNicknamePopupOpen(true);
+      queryClient.invalidateQueries([
+        'USERS',
+        {
+          id: userId,
+        },
+      ]);
+    },
   });
 
+  const handleInfoSubmit = () => {
+    patchUserInfoMutate(nickname);
+  };
+
+  const validateUserNickname = useMemo(
+    () =>
+      _.debounce((value: string) => {
+        nicknameCheck(value).then(res => {
+          res.message === '사용가능한 닉네임' ? setError('notError') : setError('isError');
```

Comment:
키 말고 값을  state  로 넣는건 어떤가요 ?
```suggestion
        nicknameCheck(value).then(res => {
          res.message !== '사용가능한 닉네임' ?  setError(errorMessage.isError) : undefined ;
```

Reply:
이렇게 리턴해주면 사용가능할때 에러메세지가 바뀌지않는데 
error 값을 undefined로 사용하자는 얘기가 맞는거죠 비비?

Reply:
맞아여!